src/utils/useHotspot.ts

```diff
@@ -186,8 +187,16 @@ const useHotspot = () => {
     await updateHotspotStatus(payload?.hotspot)
 
     if (!payload?.onboardingRecord?.onboardingKey) {
-      Logger.error(new Error('Hotspot connect failed: no onboardingKey'))
-      return handleConnectStatus('no_onboarding_key')
+      let err: HotspotConnectStatus = 'service_unavailable'
+      if (payload?.onboardingRecord?.code === 404) {
```

Comment:
does `no_onboarding_key` still occur? I see previously this returned `handleConnectStatus('no_onboarding_key')`, but now I see it returning `invalid_onboarding_address`. Additionally the two subtitle options in the i18n file are `no_onboarding_key` and `service_unavailable` but I see `invalid_onboarding_address` and `service_unavailable` being returned from here